clicky/clickyScript.py

```diff
@@ -0,0 +1,95 @@
+import requests
+import json
+from datetime import date
+from os.path import exists
+import matplotlib.pyplot as plt 
+ 
+'''
+Retrieves visitor data from the json currently checked into the
+repository.
+
+If there is no previous data, creates a template to work with 
+and returns that instead.
+
+Returns: a python dictionary of the outdated visitor data
+'''
+def getOldData():
+    if exists('clickyData.json'):
+        with open('clickyData.json') as json_file:
+            oldDataDict = json.load(json_file)
+    else:
+        oldDataDict = { 
+            'visitData': {
+                'numTotalVisits':[],
+                'numUniqueVisits':[],
+            },
+            'dates': []}
+    return oldDataDict
+        
+'''
+Retrieves the new visitor data from the Clicky url.
+
+Returns: python list containing new data from the clicky url
+'''
+def getNewData(clickyUrl):
+    # this is a bandaid solution for the SSL, requires further attention
+    # would need the certificates to be in place and removal of "verify=False"
+    response = requests.get(clickyUrl)
+    json = response.json()
+    newDataList = json[0]['dates'][0]['items']
+    return newDataList
+
+'''
+Appends the relevant parts of the new visitor data to the existing
+visitor data currently checked into the repository. This includes
+the new number of visits, new IP addresses, new unique visits, and
+the dates of the new visits.
+
+Returns: the merged data
+'''
+def merge(newData, oldData):
+    numUniqueVisits = []
```

Comment:
We might want to store the unique IP address somewhere in the stored `clickyData.json`. I don't think we necessarily care how many unique visitors visit within each week's span, but rather how many new visitors visit each week.
Otherwise put, comparing each `newData` entry to the entire list of `ipAddresses` that have ever visited the site versus just comparing it to other ipAddresses within the new week's data.

Note, this would also require changing the label in the `plotData()` function to something like "New Visits"

Reply:
I like the new logic. I tested out manually generating a graph by creating some fake data in a local `clickyData.json` file. The new graph is generated as expected and clearly portrays the data we want to gather. Thanks for updating!

Pasting graph here in case other want to review:
![clickyGraph](https://user-images.githubusercontent.com/51176978/188754474-101c68f3-f573-40d7-8f7a-4583a91bfbcb.jpg)